### DISCUSSION TITLE
Slow startup on WSL with misconfigured X server

### DISCUSSION THREAD

**Discussion by christianfosli** [helix-editor/helix#3083] [General] [answered]
I run Fedoraremix 36 distro on WSL (Windows Subsystem for Linux),
and noticed startup time for helix was super slow (ca 2 minutes).

`hx --health` and `hx --version` is fast, but opening the editor itself seemed to freeze.
After some debugging I found the culprit being `xsel -ob` timing out (in `helix_view::clipboard::get_clipboard_provider`).

I do not have an X server set up (using WSL from the terminal only on Windows 10), but for some reason the `DISPLAY` variable was set to my local IP address on Windows (by Fedoraremix tooling, see https://github.com/WhitewaterFoundry/Fedora-Remix-for-WSL/discussions/164).

If I unset the `DISPLAY` variable then helix starts super fast :smile:

i.e.
```sh
DISPLAY= hx
# starts blazingly fast!
```

instead of 
```sh
hx
# wait ~2 minutes for `xsel` to time out before showing the UI
```

---

Opening this as a discussion so hopefully if it happens to someone else they will find it and find out why :smile:.
Not sure if something should be done in helix, it is a non-helix related config error after all..

**Comment by sudormrfbin** [comment, marked as answer]
xsel has a timeout flag:

```
-t ms, --selectionTimeout ms

    Specify the timeout in milliseconds within which the selection must be retrieved.
    A value of 0 (zero) specifies no timeout (default).
```
And we could use this when checking for `xsel`:

https://github.com/helix-editor/helix/blob/d64b9abbfc9327c3041a9e8fac3c982a4d54ad99/helix-view/src/clipboard.rs#L105-L113

**Reply by christianfosli** [reply]
Good find :smile:
Unfortunately the timeout doesn't seem to apply for the issue I was experiencing:

```sh
[cfosli]$ time DISPLAY=192.168.56.1:0 xsel -ob -t 500
xsel: Can't open display: (null)
: Connection timed out

real    2m10.819s
user    0m0.001s
sys     0m0.000s
```

**Reply by sudormrfbin** [reply]
In that case i guess the only workaround would be to unset DISPLAY before starting helix or to install xclip :/

**Reply by christianfosli** [reply]
Yup, that works fine! It was just kind of annoying that DISPLAY was set incorrectly by my WSL distro. Not helix's fault :smile:
